### PR TITLE
Check event channel limit

### PR DESCRIPTION
**Pull request by cshcomcom**

### CODE REVIEW

**Comment by cshcomcom** [comment]
discussed offline) We cannot handle when lost connections, I added callback function to disconnect function. @liayoo

File: src/event-manager/event-channel-client.ts [hunk 1]
```diff
@@ -30,14 +31,32 @@ export default class EventChannelClient {
...
+            `(${eventChannelLimit})`));
+        return;
+      }
+      const numberOfEventChannels = eventHandlerNetworkInfo.numberOfEventChannels;
+      if (numberOfEventChannels === undefined) {
+        reject(new Error(`Can't get numberOfEventChannels from eventHandlerNetworkInfo ` +
+            `(${numberOfEventChannels})`));
+        return;
+      }
+      if (eventChannelLimit - numberOfEventChannels <= 0) {
```
Comment (platfowner) [inline src/event-manager/event-channel-client.ts:55]: minor) what about simply numberOfEventChannels >= eventChannelLimit ?

File: src/event-manager/event-channel-client.ts [hunk 2]
```diff
@@ -30,14 +31,32 @@ export default class EventChannelClient {
...
     return new Promise(async (resolve, reject) => {
       const eventHandlerNetworkInfo = await this._ain.net.getEventHandlerNetworkInfo();
       const url = eventHandlerNetworkInfo.url;
       if (!url) {
         reject(new Error(`Can't get url from eventHandlerNetworkInfo ` +
-            `(${JSON.stringify(eventHandlerNetworkInfo, null, 2)}`));
+            `(${JSON.stringify(eventHandlerNetworkInfo, null, 2)})`));
+        return;
       }
+      const eventChannelLimit = eventHandlerNetworkInfo.eventChannelLimit;
```
[the remark this answers is not in the document]
Comment (cshcomcom) [reply src/event-manager/event-channel-client.ts]: Yeah. I synced it! Thanks

File: src/event-manager/event-channel-client.ts (hunk 1 above)
Comment (cshcomcom) [reply src/event-manager/event-channel-client.ts]: Looks better!

File: src/event-manager/event-channel-client.ts [hunk 3]
```diff
@@ -30,14 +31,32 @@ export default class EventChannelClient {
...
+      if (eventChannelLimit === undefined) {
+        reject(new Error(`Can't get eventChannelLimit limit from eventHandlerNetworkInfo ` +
+            `(${eventChannelLimit})`));
+        return;
+      }
+      const numberOfEventChannels = eventHandlerNetworkInfo.numberOfEventChannels;
```
[the remark this answers is not in the document]
Comment (cshcomcom) [reply src/event-manager/event-channel-client.ts]: Done. Thanks!

**Comment by cshcomcom** [comment]
All comments are fixed! Now merging. Thanks for the review

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Pull request merged**